### PR TITLE
Clean up docs

### PR DESCRIPTION
**Pull request by nielslange**
In #6405, we adjusted the documentation of this repo. Due to a merge conflict, a few mistakes slipped in. This PR applies the following changes:

- Link the translation docs to the main README.md file.
- Move docs for internal developers into the corresponding subfolder.
- Delete obsolete files.

### Testing

1. Ensure that the translation docs are linked correctly.
2. Ensure that every file, that had been removed from the main docs folder, is available in a subfolder.

### CODE REVIEW

File: docs/internal-developers/testing/releases/270.md [hunk 1]
```diff
@@ -20,10 +25,13 @@ Zip file for testing: [woocommerce-gutenberg-products-block.zip](https://github.
...
-    ![Select in Hello theme](https://user-images.githubusercontent.com/3616980/84032650-f4c61700-a997-11ea-969d-6427d1e221bb.png)
--   Twenty Twenty: - Add the All Products block and the Hand-picked Products block in a page and verify (#2573): - That with the All Products block you can add the On Sale badge and it's correctly aligned in the editor and the frontend (before, it was always shown on top of the image).<br>
-    ![All Products in Twenty Twenty](https://user-images.githubusercontent.com/3616980/83013870-fef22800-a01d-11ea-8ea8-21229285d10a.png) - The Hand-picked Products block discounted prices are not underlined.<br>
+-   [Hello theme](https://elementor.com/hello-theme/):
+    -   Verify the text inside the selects is visible on hover (#2647). \
+        ![Select in Hello theme](https://user-images.githubusercontent.com/3616980/84032650-f4c61700-a997-11ea-969d-6427d1e221bb.png)
+-   Twenty Twenty:
+    -   Add the All Products block and the Hand-picked Products block in a page and verify (#2573): - That with the All Products block you can add the On Sale badge and it's correctly aligned in the editor and the frontend (before, it was always shown on top of the image). \
```
Comment (opr) [inline docs/internal-developers/testing/releases/270.md:32]: The `\` is visible in the rendered file

File: docs/internal-developers/testing/releases/270.md [hunk 2]
```diff
@@ -20,10 +25,13 @@ Zip file for testing: [woocommerce-gutenberg-products-block.zip](https://github.
...
+    -   ![All Products in Twenty Twenty](https://user-images.githubusercontent.com/3616980/83013870-fef22800-a01d-11ea-8ea8-21229285d10a.png)
+-   The Hand-picked Products block discounted prices are not underlined. \
```
Comment (opr) [inline docs/internal-developers/testing/releases/270.md:34]: This bulletpoint is not rendered underneath the image in the rendered file.

**Review by opr — CHANGES_REQUESTED** [review]
Good one Niels, but I noticed the testing notes went into `docs/internal-developers/testing` so we should update that here:
https://github.com/woocommerce/woocommerce-blocks/blob/update/clean-up-docs/.github/release-initial-checklist.md

and here:

https://github.com/woocommerce/woocommerce-blocks/blob/update/clean-up-docs/.github/patch-initial-checklist.md

File: docs/internal-developers/testing/releases/270.md (hunk 1 above)
Comment (nielslange) [reply docs/internal-developers/testing/releases/270.md:32]: When looking at https://github.com/woocommerce/woocommerce-blocks/blob/fad6a96e657d84759c1ccb4346e3077b747bc3a3/docs/internal-developers/testing/releases/270.md, I do not see the `\` rendered.

Comment (nielslange) [reply docs/internal-developers/testing/releases/270.md:32]: Good catch. The problem was caused by the list item in the next row. This should not be a separate list item, but belong to the previous one.

File: docs/internal-developers/testing/releases/270.md (hunk 2 above)
Comment (nielslange) [reply docs/internal-developers/testing/releases/270.md:34]: This is the same problem as mentioned in https://github.com/woocommerce/woocommerce-blocks/pull/6550#discussion_r901408862. The image should not be in a separate list item, but belongs to the previous one. I've corrected this problem.

**Comment by nielslange** [comment]
@opr I've addressed all points you mentioned. Would you mind reviewing this PR again?

**Review by opr — APPROVED** [review]
Thanks Niels! Looks good now! 🚢 it 🥳

**Pull request merged**